spark/src/main/java/org/opensearch/sql/spark/rest/model/CreateAsyncQueryRequest.java

```diff
@@ -8,36 +8,43 @@
 import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
 
 import java.io.IOException;
-import lombok.AllArgsConstructor;
 import lombok.Data;
+import org.apache.commons.lang3.Validate;
 import org.opensearch.core.xcontent.XContentParser;
 
 @Data
-@AllArgsConstructor
 public class CreateAsyncQueryRequest {
 
   private String query;
+  private String datasource;
   private LangType lang;
 
+  public CreateAsyncQueryRequest(String query, String datasource, LangType lang) {
```

Comment:
should we add datasource for ALL the async_query? then, Plugin AuthZ against (1) async_query API (2) datasource usage permission.

Reply:
Yes we need to add all the async queries.
We perform both the permission checks. In SparkQueryDispatcher we perform datasource Authorization.

Reply:
as discussed offline, no need to addd datasource for GET result and Cancel.